includes/wf_crm_webform_postprocess.inc

```diff
@@ -2406,11 +2406,22 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
               $createModeKey = 'civicrm_' . $c . '_contact_' . $n . '_' . $table . '_createmode';
               $multivaluesCreateMode = $this->data['config']['create_mode'][$createModeKey] ?? NULL;
               $cgMaxInstance = $this->all_sets[$table]['max_instances'] ?? 1;
-              if ($cgMaxInstance > 1 && empty($multivaluesCreateMode) && !empty($this->existing_contacts[$c])) {
-                $existingValue = $this->getCustomData($this->existing_contacts[$c], NULL, FALSE)[$table] ?? NULL;
-                if (!empty($existingValue)) {
-                  $name = "{$name}_" . key($existingValue);
+              // Is this a multi-value custom field?
+              if ($cgMaxInstance > 1) {
+                // For edit mode, update the keys to 'custom_<fieldID>_<existing_value_id>.
```

Comment:
@KarinG Have tested this on a complex webform and seems like there's an extra line required here to initialize the value of $existingValue.

    $existingValue = NULL;

To replicate the issue:

- Select 4 in the number of multi-valued fields on the webform
- Select first 2 as `Create/Edit` and other 2 as `Create Only`.
- Submit the webform. It correctly updates the first 2 values but "creates" only 1 from the other 2 fields.

Reply:
Thank you for testing @jitendrapurohit 👍

🤔 in both 7.x-5.x as well as 8.x-5.x we have a `?? NULL` already. So not sure why/where this is needed?

```
$existingValue = $this->getCustomData($this->existing_contacts[$c], NULL, FALSE)[$table] ?? NULL;
if (!empty($existingValue) && is_array($existingValue)) {
   existingValue = array_slice($existingValue, $n -1, 1, TRUE);
   $name = "{$name}_" . key($existingValue);
}
```

Reply:
@KarinG This line is required before entering the if statement, i.e, before line 2412. This is also needed in the d8 PR. Since that is already merged, it would need a separate PR for the fix.